subsys/net/ip/icmp.c

```diff
@@ -358,6 +384,45 @@ static int send_icmpv6_echo_request(struct net_icmp_ctx *ctx,
 }
 #endif
 
+static int get_offloaded_ping_handler(struct net_if *iface,
+				      net_icmp_offload_ping_handler_t *ping_handler)
+{
+	struct net_icmp_offload *offload;
+	int ret;
+
+	if (!IS_ENABLED(CONFIG_NET_OFFLOAD)) {
+		return -ENOTSUP;
+	}
+
+	if (iface == NULL) {
+		return -EINVAL;
+	}
+
+	if (!net_if_is_ip_offloaded(iface)) {
+		return -ENOENT;
+	}
+
+	ret = 0;
+
+	k_mutex_lock(&lock, K_FOREVER);
+
+#if defined(CONFIG_NET_OFFLOAD)
+	SYS_SLIST_FOR_EACH_CONTAINER(&offload_handlers, offload, node) {
+		if (offload->iface == iface) {
+			ping_handler = &offload->ping_handler;
```

Comment:
Hmm, we overwrite local variable (pointer) here, shouldn't it actually be dereferenced before write?

Reply:
Hmm, I am not following you here. We are returning the ping_handler to the caller so that is why the code looks like this.

Reply:
I think we're on the same page now, the latest push illustrates best what I meant here:
```patch
-			ping_handler = &offload->ping_handler;
+			*ping_handler = offload->ping_handler;
```

Reply:
Ah yes, I had fixed that earlier before your comment but before sending that version to review so got a bit confused.